### PR TITLE
Changing Llama Guard safety check to HF classes

### PR DESCRIPTION
**Pull request by albertodepaola**
# What does this PR do?
Changes the Llama Guard safety check to use HF classes. Removes the previously used code to load Llama Guard.

## Feature/Issue validation/testing

* Llama guard enabled:
* Command: `python examples/inference.py --model_name ../llama/models_hf/7B --prompt_file examples/test_prompt_2.txt --enable_llamaguard_content_safety --enable_salesforce_content_safety False`
* Result
image

* Llama Guard disabled:
* Command: `python examples/inference.py --model_name ../llama/models_hf/7B --prompt_file examples/test_prompt_2.txt`
* Result
image

* Salesforce, Llama Guard and sensitive topics enabled:
* Command: `python examples/inference.py --model_name ../llama/models_hf/7B --prompt_file examples/test_prompt_2.txt --enable_llamaguard_content_safety --enable_sensitive_topics`
* Result
image

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/llama-recipes/blob/main/CONTRIBUTING.md#pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes?  
- [ ] Did you write any new necessary tests?

Thanks for contributing 🎉!

### CODE REVIEW

File: src/llama_recipes/inference/safety_utils.py [hunk 1]
```diff
@@ -92,7 +91,7 @@ def get_total_length(self, data):
 
 # Class for performing safety checks using Azure Content Safety service
 class AzureSaftyChecker(object):
-    def __init__(sel, **kwargs):
+    def __init__(sel):
```
Comment (mreso) [inline src/llama_recipes/inference/safety_utils.py:94]: Not part of this PR but self is mistyped as sel but then references as self

**Review by mreso — APPROVED** [review]
Marked a bug in the comments which was already there but we should still fix it, otherwise LGTM

**Review by HamidShojanazeri — COMMENTED** [review]
@albertodepaola thanks for the PR, I wonder if we need a separate example? `llama_gurda_inference.py`. 

Also, I wonder if still are keeping the llama_guard folder? if so I suggest to move this example there as it seems we still having the prompt_formatter there.

Please also add a llama_guard announcement on the main REAMDE and add a link to the respective folder/example.

**Comment by albertodepaola** [comment]
@HamidShojanazeri The reference to the Llama Guard scripts in the relevant Readmes! 
@QingHu1227 Referencing the prompt formatting methods in the stand alone script only. For the safety checker, I'm using the method provided by the tokenizer. We can evaluate adding custom categories to that as well in the future.

**Review by QingHu1227 — APPROVED** [review]
The prompt format change looks good to me.
